### PR TITLE
WIP: apiserver: log trace and span ID

### PR DESCRIPTION
**Pull request by pohly**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

This enables contextual logging for one log call of the apiserver (just as a proof-of-concept) and uses new logr APIs to automatically log the trace and span ID if the context has them.

#### Which issue(s) this PR fixes:
Related-to https://github.com/kubernetes/klog/issues/356, https://github.com/kubernetes/kubernetes/issues/113414

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/server/options/tracing.go [hunk 1]
```diff
@@ -123,9 +126,34 @@ func (o *TracingOptions) ApplyTo(es *egressselector.EgressSelector, c *server.Co
...
+func traceIDsFromContext(ctx context.Context) logr.KeysAndValues {
+	spanContext := trace.SpanContextFromContext(ctx)
+	if !spanContext.IsValid() {
+		return nil
+	}
+
+	// A valid span context has both IDs set. These key names are
+	// determined by
+	// https://github.com/open-telemetry/opentelemetry-specification/tree/a87fb8c5a60f942d08c67c1582034eb8b4322009/specification/logs#trace-context-in-legacy-formats
+	return logr.KeysAndValues{
```
Comment (thockin) [inline staging/src/k8s.io/apiserver/pkg/server/options/tracing.go:151]: would we want to return this as `"span", struct { trace_id, span_id string}{...}` or something, so they render as a struct?

Comment (pohly) [reply staging/src/k8s.io/apiserver/pkg/server/options/tracing.go:151]: No, @dashpole wanted this to follow the OpenTelemetry spec, which implies that the IDs need to be at the top level, not embedded in some struct.

**Comment by dims** [comment]
This PR has the label work-in-progress, please revisit to see if you still need this, please close if not

**Comment by pohly** [comment]
/close

This was only a proof-of-concept.

**Pull request closed without merging**